app/controllers/redirection_controller.rb

```diff
@@ -1,28 +1,41 @@
-                  level_one_taxon: params['taxons'].try(:first),
-                  level_two_taxon: params['subtaxons'].try(:first),
-                  people: params['people'],
-                  organisations: params['departments'],
-                  world_locations: params['world_locations'],
-                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
     respond_to do |format|
-      format.html { redirect_to(finder_path('search/news-and-communications', params: parameters)) }
-      format.atom { redirect_to(finder_path('search/news-and-communications', params: parameters, format: :atom)) }
+      format.html { redirect_to(finder_path('search/news-and-communications', params: convert_common_parameters)) }
+      format.atom { redirect_to(finder_path('search/news-and-communications', params: convert_common_parameters, format: :atom)) }
     end
   end
 
   def publications
-    parameters = { keywords: params['keywords'],
-                  level_one_taxon: params['taxons'].try(:first),
-                  level_two_taxon: params['subtaxons'].try(:first),
-                  organisations: params['departments'],
-                  world_locations: params['world_locations'],
-                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
     respond_to do |format|
-      format.html { redirect_to(finder_path('search/all', params: parameters)) }
-      format.atom { redirect_to(finder_path('search/all', params: parameters, format: :atom)) }
+      format.html { redirect_to(finder_path('search/all', params: convert_common_parameters)) }
+      format.atom { redirect_to(finder_path('search/all', params: convert_common_parameters, format: :atom)) }
     end
   end
+
+  def published_statistics
+    respond_to do |format|
+      format.html { redirect_to(finder_path('search/statistics', params: convert_common_parameters)) }
+      format.atom { redirect_to(finder_path('search/statistics', params: convert_common_parameters, format: :atom)) }
+    end
+  end
+
+  def upcoming_statistics
+    respond_to do |format|
+      format.html { redirect_to(finder_path('search/statistics', params: convert_common_parameters.merge(content_store_document_type: :statistics_upcoming))) }
+      format.atom { redirect_to(finder_path('search/statistics', params: convert_common_parameters.merge(content_store_document_type: :statistics_upcoming), format: :atom)) }
+    end
+  end
+
+private
+
+  def convert_common_parameters
+    { keywords: params['keywords'],
+      level_one_taxon: params['taxons'].try(:first) || params['topics'].try(:first),
+      level_two_taxon: params['subtaxons'].try(:first),
+      organisations: params['departments'] || params['organisations'],
+      people: params['people'],
+      world_locations: params['world_locations'],
+      public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
```

Comment:
You'll need to call `.compact` on the final `}` otherwise you'll have empty keys being passed in.

Reply:
Cool, made the change now

Reply:
:shipit: